silo/api.py

```diff
@@ -145,18 +145,28 @@ def create(self, request, *args, **kwargs):
         if not request.user.is_superuser:
             return Response(status=status.HTTP_403_FORBIDDEN)
 
+        # serialize and validate data
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        # get fields' values
+        form_uuid = serializer.data['form_uuid']
+        level1_uuid = serializer.data['level1_uuid']
+        tola_user_uuid = serializer.data['tola_user_uuid']
+        form_name = serializer.data['name']
+        read_name = serializer.data['name']
+        columns = serializer.data['fields']
+        description = serializer.data.get('description', '')
+
+        # we need to convert the dict into a JSON to be stored
+        columns = json.dumps(columns)
+
         try:
-            form_uuid = request.POST['form_uuid']
-            level1_uuid = request.POST['level1_uuid']
-            tola_user_uuid = request.POST['tola_user_uuid']
             wkflvl1 = WorkflowLevel1.objects.get(level1_uuid=level1_uuid)
             tola_user = TolaUser.objects.get(tola_user_uuid=tola_user_uuid)
-            form_name = request.POST['name']
-            read_name = request.POST['name']
-            columns = request.POST['fields']
-        except (WorkflowLevel1.DoesNotExist, TolaUser.DoesNotExist, KeyError) \
-                as e:
-            return Response(e, status=status.HTTP_400_BAD_REQUEST)
+        except (WorkflowLevel1.DoesNotExist, TolaUser.DoesNotExist) as e:
+            return Response(
+                e.message, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
We should better leave 500 errors for system crashes, not for something we may expect and can control.

Reply:
I agree. but what do you suggest?

Reply:
Leave 400, as you did. That was OK.